gui/wxpython/startup/guiutils.py

```diff
@@ -279,44 +280,66 @@ def rename_location_interactively(guiparent, grassdb, location):
     return newlocation
 
 
-def delete_mapset_interactively(guiparent, grassdb, location, mapset):
+def delete_mapset_interactively(guiparent, db_loc_mapset):
     """
-    Delete selected mapset
+    Delete selected mapset/s. Whether PERMANENT or current mapsets found,
+    delete operation is not performed.
     """
-    if mapset == "PERMANENT":
-        GMessage(
+    genv = gisenv()
+
+    # Initialization
+    issues = []
+
+    # Check selected mapsets and remember issues
+    for grassdb, location, mapset in db_loc_mapset:
+
+        # Check for permanent mapsets
+        if mapset == "PERMANENT":
+            issues.append(os.path.join(grassdb, location, mapset) +
+                          " is required for a valid location.")
+        # Check for current mapsets
+        if (grassdb == genv['GISDBASE'] and
+                location == genv['LOCATION_NAME'] and
+                mapset == genv['MAPSET']):
+            issues.append(os.path.join(grassdb, location, mapset) +
+                          " is a current mapset.")
+
+    # If any issues, display the warning message and do not delete anything
+    if issues:
+        reasons = "\n".join(issues)
+        dlg = wx.MessageDialog(
             parent=guiparent,
             message=_(
-                "Mapset <PERMANENT> is required for valid GRASS location.\n\n"
-                "This mapset cannot be deleted."
-            ),
+                "Cannot delete one or more mapsets for the following reasons:\n\n"
+                "%(reasons)s"
+            )
+            % {"reasons": reasons},
+            caption=_("Delete selected mapsets"),
+            style=wx.OK | wx.ICON_WARNING
         )
-        return
-
-    dlg = wx.MessageDialog(
-        parent=guiparent,
-        message=_(
-            "Do you want to continue with deleting mapset <%(mapset)s> "
-            "from location <%(location)s>?\n\n"
-            "ALL MAPS included in this mapset will be "
-            "PERMANENTLY DELETED!"
+        dlg.ShowModal()
+    else:
+        dlg = wx.MessageDialog(
+            parent=guiparent,
+            message=_(
+                "Do you want to continue with deleting one or more mapsets?\n\n"
+                "ALL MAPS included in these mapsets will be "
```

Comment:
This should lit the mapsets that will be deleted.

Reply:
It lists all mapsets that will be deleted in terms of their paths. 
![Screenshot from 2020-07-23 05-01-48](https://user-images.githubusercontent.com/49241681/88274700-6655ec80-ccdc-11ea-86de-477daf6fab61.png)